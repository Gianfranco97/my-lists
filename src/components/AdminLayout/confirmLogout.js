import { Modal } from "antd";

const { confirm } = Modal;

function showDeleteConfirm(onOk) {
  confirm({
    title: "Are you sure you want to close the session?",
    okText: "Yes",
    okType: "danger",
    cancelText: "No",
    onOk() {
      onOk();
    },
  });
}

export default showDeleteConfirm;
