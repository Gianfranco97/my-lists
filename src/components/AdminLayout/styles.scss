.admin-layout {
  height: 100%;

  .logo {
    padding: 15px;

    img {
      width: 160px;
    }
  }

  .admin-content {
    overflow: auto;
  }

  .site-layout-sub-header-background {
    background: white;
  }

  .site-layout-background {
    background: white;
  }

  Footer {
    text-align: center;
  }
}
