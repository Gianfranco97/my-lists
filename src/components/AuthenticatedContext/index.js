import React, { useState } from "react";

const AuthenticatedContext = React.createContext();

export function AuthenticatedProvider(props) {
  const [isAuthenticated, setIsAuthenticated] = useState(
    !!sessionStorage.getItem("session-token")
  );

  return (
    <AuthenticatedContext.Provider
      {...props}
      value={{
        isAuthenticated: isAuthenticated,
        changeAuthenticatedStatus: setIsAuthenticated,
      }}
    />
  );
}

export function useAuthenticatedContext() {
  const context = React.useContext(AuthenticatedContext);

  if (!context) {
    throw new Error(
      "'useAuthenticatedContext' must be inside the 'AuthenticatedProvider"
    );
  }

  return context;
}

export default AuthenticatedContext;
