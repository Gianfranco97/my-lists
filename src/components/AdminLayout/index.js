import React from "react";
import { Layout, Menu } from "antd";
import {
  LogoutOutlined,
  HomeOutlined,
} from "@ant-design/icons";
import { withRouter } from "react-router-dom";
import PropTypes from "prop-types";
import confirmLogout from "./confirmLogout";
import AuthenticatedContext from "../AuthenticatedContext";

import "./styles.scss";

const { Header, Content, Footer, Sider } = Layout;

class AdminLayout extends React.Component {
  handleClick = (url) => {
    const { history } = this.props;
    history.push(url);
  };

  logout = (changeAuthenticatedStatus) => {
    const { history } = this.props;

    confirmLogout(async () => {
      try {
        sessionStorage.clear();
        changeAuthenticatedStatus(false);
        history.push("/login");
      } catch (error) {}
    });
  };

  render() {
    const { children, title } = this.props;

    return (
      <Layout className="admin-layout">
        <Sider breakpoint="lg" collapsedWidth="0">

          <AuthenticatedContext.Consumer>
            {({ changeAuthenticatedStatus }) => (
              <Menu theme="dark" mode="inline" defaultSelectedKeys={[title]}>
                <Menu.Item
                  onClick={() => this.handleClick("/")}
                  key="Home"
                  icon={<HomeOutlined />}
                >
                  My lists
                </Menu.Item>

                <Menu.Item
                  key="4"
                  icon={<LogoutOutlined />}
                  onClick={() => this.logout(changeAuthenticatedStatus)}
                >
                  Logout
                </Menu.Item>
              </Menu>
            )}
          </AuthenticatedContext.Consumer>
        </Sider>
        <Layout>
          <Header
            className="site-layout-sub-header-background"
            style={{ padding: 0 }}
          >
            <h1 style={{ marginLeft: 20 }}>{title}</h1>
          </Header>
          <Content className="admin-content" style={{ margin: "24px 16px 0" }}>
            <div className="site-layout-background" style={{ padding: 24 }}>
              {children}
            </div>
          </Content>
          <Footer style={{ textAlign: "center" }}>
            Gianfranco Manganiello Â©2020
          </Footer>
        </Layout>
      </Layout>
    );
  }
}

AdminLayout.propTypes = {
  history: PropTypes.object.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
    PropTypes.array,
  ]).isRequired,
};

export default withRouter(AdminLayout);
