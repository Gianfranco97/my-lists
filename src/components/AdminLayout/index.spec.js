import React from "react";
import { shallow } from "enzyme";
import AdminLayout from "./index.js";
import { MemoryRouter } from "react-router-dom";
import renderer from "react-test-renderer";
import { AuthenticatedProvider } from "../AuthenticatedContext";

it("The snapshot matches", () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <AuthenticatedProvider>
          <AdminLayout>
            <div />
          </AdminLayout>
        </AuthenticatedProvider>
      </MemoryRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("renders without crashing", () => {
  shallow(
    <AdminLayout>
      <div />
    </AdminLayout>
  );
});
