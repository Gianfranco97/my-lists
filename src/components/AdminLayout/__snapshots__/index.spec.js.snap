// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The snapshot matches 1`] = `
<section
  className="ant-layout ant-layout-has-sider admin-layout"
>
  <aside
    className="ant-layout-sider ant-layout-sider-dark"
    style={
      Object {
        "flex": "0 0 200px",
        "maxWidth": "200px",
        "minWidth": "200px",
        "width": "200px",
      }
    }
  >
    <div
      className="ant-layout-sider-children"
    >
      <ul
        className="ant-menu ant-menu-dark ant-menu-root ant-menu-inline"
        onMouseEnter={[Function]}
        onTransitionEnd={[Function]}
        role="menu"
        style={Object {}}
      >
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <span
            aria-label="home"
            className="anticon anticon-home"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="home"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M946.5 505L560.1 118.8l-25.9-25.9a31.5 31.5 0 00-44.4 0L77.5 505a63.9 63.9 0 00-18.8 46c.4 35.2 29.7 63.3 64.9 63.3h42.5V940h691.8V614.3h43.4c17.1 0 33.2-6.7 45.3-18.8a63.6 63.6 0 0018.7-45.3c0-17-6.7-33.1-18.8-45.2zM568 868H456V664h112v204zm217.9-325.7V868H632V640c0-22.1-17.9-40-40-40H432c-22.1 0-40 17.9-40 40v228H238.1V542.3h-96l370-369.7 23.1 23.1L882 542.3h-96.1z"
              />
            </svg>
          </span>
          <span>
            My lists
          </span>
        </li>
        <li
          className="ant-menu-item"
          onClick={[Function]}
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          role="menuitem"
          style={
            Object {
              "paddingLeft": 24,
            }
          }
        >
          <span
            aria-label="logout"
            className="anticon anticon-logout"
            role="img"
          >
            <svg
              aria-hidden="true"
              className=""
              data-icon="logout"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M868 732h-70.3c-4.8 0-9.3 2.1-12.3 5.8-7 8.5-14.5 16.7-22.4 24.5a353.84 353.84 0 01-112.7 75.9A352.8 352.8 0 01512.4 866c-47.9 0-94.3-9.4-137.9-27.8a353.84 353.84 0 01-112.7-75.9 353.28 353.28 0 01-76-112.5C167.3 606.2 158 559.9 158 512s9.4-94.2 27.8-137.8c17.8-42.1 43.4-80 76-112.5s70.5-58.1 112.7-75.9c43.6-18.4 90-27.8 137.9-27.8 47.9 0 94.3 9.3 137.9 27.8 42.2 17.8 80.1 43.4 112.7 75.9 7.9 7.9 15.3 16.1 22.4 24.5 3 3.7 7.6 5.8 12.3 5.8H868c6.3 0 10.2-7 6.7-12.3C798 160.5 663.8 81.6 511.3 82 271.7 82.6 79.6 277.1 82 516.4 84.4 751.9 276.2 942 512.4 942c152.1 0 285.7-78.8 362.3-197.7 3.4-5.3-.4-12.3-6.7-12.3zm88.9-226.3L815 393.7c-5.3-4.2-13-.4-13 6.3v76H488c-4.4 0-8 3.6-8 8v56c0 4.4 3.6 8 8 8h314v76c0 6.7 7.8 10.5 13 6.3l141.9-112a8 8 0 000-12.6z"
              />
            </svg>
          </span>
          <span>
            Logout
          </span>
        </li>
      </ul>
    </div>
  </aside>
  <section
    className="ant-layout"
  >
    <header
      className="ant-layout-header site-layout-sub-header-background"
      style={
        Object {
          "padding": 0,
        }
      }
    >
      <h1
        style={
          Object {
            "marginLeft": 20,
          }
        }
      />
    </header>
    <main
      className="ant-layout-content admin-content"
      style={
        Object {
          "margin": "24px 16px 0",
        }
      }
    >
      <div
        className="site-layout-background"
        style={
          Object {
            "padding": 24,
          }
        }
      >
        <div />
      </div>
    </main>
    <footer
      className="ant-layout-footer"
      style={
        Object {
          "textAlign": "center",
        }
      }
    >
      Gianfranco Manganiello Â©2020
    </footer>
  </section>
</section>
`;
